src/server/ua_server.c

```diff
@@ -58,23 +58,22 @@ UA_Server_forEachChildNodeCall(UA_Server *server, UA_NodeId parentNodeId,
      * delete references from within the callback. In single-threaded mode this
      * changes the same node we point at here. In multi-threaded mode, this
      * creates a new copy as nodes are truly immutable. */
-    UA_ReferenceNode *refs = NULL;
-    size_t refssize = parent->referencesSize;
-    UA_StatusCode retval = UA_Array_copy(parent->references, parent->referencesSize,
-        (void**)&refs, &UA_TYPES[UA_TYPES_REFERENCENODE]);
+    UA_NodeReferenceKind *refs = NULL;
+    UA_StatusCode retval = UA_NodeReferenceKind_copy(parent->references, parent->referencesSize, &refs);
```

Comment:
So the problem happens when the iterate-callback removes references from the node itself?
The code would be simpler if we just made a copy of the entire node. :-)
And then continue to point to the nodeids in-place.

Reply:
I just looked again at the code, do we even need this ugly copy?
The callback in line 71 gets only node ids. And we should set the callback parameters to const

Reply:
The callback could remove a node via the regular public API.
This can remove a member of the nodes-array we iterate over...

Reply:
Ah, right. Got it, so I will just create a copy of parent and use that copy for iteration